### DISCUSSION TITLE
apollo-datasource with express ?

### DISCUSSION THREAD

**Discussion by Rahmo** [apollographql/apollo-server#5031] [Q&A]
Hi I just built an Apollo-Server-Lambda with data sources - I like how everything is abstracted right there. 

I have another app that uses Apollo-server-express 
and I wonder if I can pass down data sources instead of context 
as within data sources I will have a source for apis and another for db 

any thought ? 

Thanks

**Comment by Rahmo** [comment]


**Reply by glasser** [reply]
Data sources should work with all the Apollo Server integrations and are not Lambda-specific at all. Did you have trouble applying the same approach that worked for you with Lambda to Express?

**Reply by glasser** [reply]
Hi! As a maintainer of AS I'm not really able to dig in deep to helping you use the software, but I hope others on this discussion area can help with that. I'm much better at looking at a complete reproduction (like something I can `git clone` or look at on codesandbox.io) that directly demonstrates a bug that I can see on my machine.

If the issue is literally "feeds" is undefined, since the only place that "feeds" is mentioned is in `this.prisma.feeds`, then that sounds like it's something relating to how you set up your prisma object, something I know nothing about.

**Reply by Rahmo** [reply]
alright thanks
